File: proxy_server.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import requests
import base64
import io
from PIL import Image

app = Flask(__name__)
CORS(app)

@app.route('/predict', methods=['POST'])
def predict():
    try:
        # Get the image data from the request
        data = request.json
        if not data or 'image' not in data:
            return jsonify({'error': 'No image data provided'}), 400

        # Decode base64 image
        image_data = base64.b64decode(data['image'])
        
        # Create a file-like object from the image data
        image_file = io.BytesIO(image_data)
        
        # Create a FormData object with the correct field name
        files = {'file': ('image.jpg', image_file, 'image/jpeg')}
        
        # Forward the request to the cloud function
        response = requests.post(
            'https://us-central1-aurora-457407.cloudfunctions.net/predict',
            files=files,
            headers={
                'Accept': 'application/json'
            }
        )
        
        # Check if the response is successful
        if response.status_code != 200:
            return jsonify({'error': f'Cloud function error: {response.text}'}), response.status_code
        
        # Return the response from the cloud function
        return jsonify(response.json()), 200
        
    except Exception as e:
        print(f"Error in proxy server: {str(e)}")
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(port=5000, debug=True) 

File: proxy_server.js
const express = require('express');
const cors = require('cors');
const multer = require('multer');
const axios = require('axios');
const FormData = require('form-data');

const app = express();
const upload = multer();

// Enable CORS for all routes
app.use(cors());
app.use(express.json());

// Serve static files
app.use(express.static('.'));

app.post('/predict', upload.single('file'), async (req, res) => {
    try {
        if (!req.file) {
            return res.status(400).json({ error: 'No file uploaded' });
        }

        // Create form data
        const formData = new FormData();
        formData.append('file', req.file.buffer, {
            filename: 'image.jpg',
            contentType: 'image/jpeg'
        });

        // Forward to cloud function
        const response = await axios.post(
            'https://us-central1-aurora-457407.cloudfunctions.net/predict',
            formData,
            {
                headers: {
                    ...formData.getHeaders(),
                    'Accept': 'application/json'
                }
            }
        );

        res.json(response.data);
    } catch (error) {
        console.error('Error:', error.message);
        res.status(500).json({ error: error.message });
    }
});

const PORT = 3000;
app.listen(PORT, () => {
    console.log(`Server running at http://localhost:${PORT}`);
}); 